### ISSUE TITLE
Ussage with WSL

### ISSUE THREAD

**Issue by Zelinzky** [FalsePattern/ZigBrains#55]
The plugin works relatively well, but when working with wsl projects it continues to use windows zig binaries.
It would be ideal that the plugin detects when running a wsl project and uses zig and zls from the wsl path.

**Comment by FalsePattern** [comment]
You can change the zig and zls paths to point to the "wsl" binaries in the per-project settings under `Settings -> Languages & Frameworks -> Zig`; `Zig Settings -> Toolchain location` and `ZLS launch settings -> Executable path`

**Comment by Zelinzky** [comment]
yes, i attempted to do this. I point to zls binary no problem, but whenever i select my instalation path for zig itself it is ignored.
![Screenshot 2024-07-23 115730](https://github.com/user-attachments/assets/cf63afa6-9bb9-4d4b-8ccd-c0a7c1a4662c)

After i hit apply and reopen the config it is gone. and it never shows the toolchain version like it does with the windows binaries.

![image](https://github.com/user-attachments/assets/258a96f8-40ef-4d91-8768-201d66381921)

**Comment by FalsePattern** [comment]
ZigBrains can only run native executables for zig and zls at the moment, support for cross-platform and remote binaries is planned in the future, but for now you'll need to use a native windows exe for them (or alternatively, install a linux version of your IDE into the WSL environment, and then use intellij Code With Me to remotely connect to it).

**Comment by Zelinzky** [comment]
Gotcha!

**Comment by cmcpasserby** [comment]
@Zelinzky how are you running things i have been using it with WSL just fine, since its running on the remote side using the gateway stuff in clion. works almost as well as when i am booted into a proper linux distro.
